test/integration/not-ready-test.js

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var test = require('tape');
+var _ = require('underscore');
+var async = require('async');
+
+var testRingpopCluster = require('../lib/test-ringpop-cluster.js');
+var sendPing = require('../../lib/gossip/ping-sender.js');
+var sendPingReqs = require('../../lib/gossip/ping-req-sender.js');
+var Member = require('../../lib/membership/member.js');
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 3,
+    tap: function(cluster) {
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'not-ready error on ping', function t(bootRes, cluster, assert) {
+    assert.plan(1);
+
+    var ringpop = cluster[3];
```

Comment:
That's a question I wanted you to answer in the comments. "Why are we manually adding the member to the membership with makeAlive?" aka "What's the purpose?"